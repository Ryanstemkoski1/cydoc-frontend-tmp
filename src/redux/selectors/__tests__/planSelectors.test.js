import {
    flattenConditionCategory,
    // selectPlanCondition,
} from '../planSelectors';
// import {
//     initialPlan,
//     conditionId,
// } from 'screens/EditNote/content/discussionplan/util';
// import { initialPlanState } from '../../reducers/planReducer';

describe('plan selectors', () => {
    it('flattenConditionCategory includes id along with original keys', () => {
        const diagnosis = {
            diagnosis: 'foo',
            comments: 'bar',
        };
        const id = '000';

        expect(flattenConditionCategory([id, diagnosis])).toEqual({
            id,
            ...diagnosis,
        });
    });

    // // TODO: Fix below tests
    // it('selectPlanCondition correctly flattens all objects', () => {
    //     const state = { plan: initialPlan };
    //     expect(selectPlanCondition(state, conditionId)).toMatchSnapshot();
    // });

    // it('selectPlanCondition uses values from initialPlan if id is invalid', () => {
    //     const id = Object.keys(initialPlanState.conditions)[0];
    //     const state = { plan: initialPlanState };
    //     const invalidId = 'FAKEEE ID';

    //     const initialCondition = initialPlanState.conditions[id];
    //     const diagnosisId = Object.keys(
    //         initialCondition.differentialDiagnoses
    //     )[0];
    //     const prescriptionId = Object.keys(initialCondition.prescriptions)[0];
    //     const procedureId = Object.keys(
    //         initialCondition.proceduresAndServices
    //     )[0];
    //     const referralId = Object.keys(initialCondition.referrals)[0];

    //     // everything but the id should come from the initialPlanState
    //     expect(selectPlanCondition(state, invalidId)).toEqual({
    //         ...initialCondition,
    //         id: invalidId,
    //         differentialDiagnoses: [
    //             {
    //                 id: diagnosisId,
    //                 ...initialCondition.differentialDiagnoses[diagnosisId],
    //             },
    //         ],
    //         prescriptions: [
    //             {
    //                 id: prescriptionId,
    //                 ...initialCondition.prescriptions[prescriptionId],
    //             },
    //         ],
    //         proceduresAndServices: [
    //             {
    //                 id: procedureId,
    //                 ...initialCondition.proceduresAndServices[procedureId],
    //             },
    //         ],
    //         referrals: [
    //             { id: referralId, ...initialCondition.referrals[referralId] },
    //         ],
    //     });
    // });
});
