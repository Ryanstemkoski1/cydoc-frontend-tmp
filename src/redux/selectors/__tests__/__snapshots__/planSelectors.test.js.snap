// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plan selectors selectPlanCondition correctly flattens all objects 1`] = `
Object {
  "differentialDiagnoses": Array [
    Object {
      "comments": "",
      "diagnosis": "",
      "id": "category",
    },
  ],
  "id": "condition",
  "name": "",
  "prescriptions": Array [
    Object {
      "comments": "",
      "dose": "",
      "id": "category",
      "signature": "",
      "type": "",
    },
  ],
  "proceduresAndServices": Array [
    Object {
      "comments": "",
      "id": "category",
      "procedure": "",
      "when": "",
    },
  ],
  "referrals": Array [
    Object {
      "comments": "",
      "department": "",
      "id": "category",
      "when": "",
    },
  ],
}
`;
